### ISSUE TITLE
Records with parameters can't contain iterator methods

### ISSUE THREAD

**Issue by Happypig375** [dotnet/roslyn#46657]
**Version Used**: latest master (45f2aedcfe54c9b0cc6d9255e3b8945942359f1b)

**Steps to Reproduce**:

https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEAMACFBGALAbgFgAoAOwjDigAcIBjOLbAOgCUBXUmAS0uYGEA9mGrcANogDKiAG7cGUDAG8sAJgwBJKAFEAHvATkxG0txjKMAXyskSKAMwYEcOoIQATDAA0AFNy4YEACUJEokGBFYjjj2ADz+MAB8GADicDAa8GBQPiHEkRY42FgA7BjY+EzFKGWqlZYkDcR20eoAsqHhkQ5McQnJaRlZOXkFKkWl5ZUTNRh1NsSWQA==
```cs
using System.Collections.Generic;
namespace System.Runtime.CompilerServices { class IsExternalInit { } }

public record X(int a)
{
    public IEnumerable<int> GetItems()
    { yield return 1; yield return 2; }
}

public class M
{
    public IEnumerable<int> GetItems()
    { yield return 1; yield return 2; }
}
```

**Expected Behavior**: Compiles fine.

**Actual Behavior**: `error CS1624: The body of 'X.X(int)' cannot be an iterator block because 'void' is not an iterator interface type`

**Comment by YairHalberstadt** [comment]
Fix available at #46713

**Comment by jcouv** [comment]
Thanks @Happypig375 for reporting this and @YairHalberstadt for fixing it!
